sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -597,11 +606,13 @@ static String getExtractDestinationUri(String extractDestinationDir) {
     }
     long filesCount = counts.get(0);
 
-    ImmutableList.Builder<String> paths = ImmutableList.builder();
-    IOChannelFactory factory = IOChannelUtils.getFactory(extractDestinationDir);
+    ImmutableList.Builder<ResourceId> paths = ImmutableList.builder();
+    ResourceId extractDestinationDirResourceId =
+        FileSystems.matchSingleFileSpec(extractDestinationDir).resourceId();
```

Comment:
I think at this point we expect the extract dir to exist and fail if it doesn't? (This isn't a deviation from the existing behavior)